jujugui/static/gui/src/app/components/added-services-list-item/test-added-services-list-item.js

```diff
@@ -248,4 +387,185 @@ describe('AddedServicesListItem', function() {
     });
   });
 
+  it('correctly sets the visibility icons status on render', () => {
+    var service = {
+      getAttrs: function() {
+        return {
+          icon: 'icon.gif', unit_count: '2', name: 'demo', id: 'demo',
+          units: {
+            toArray: function() {
+              return [{agent_state: 'uncommitted'}, {agent_state: 'pending'}];
+            }}};
+      },
+      get: function() {
+        return true;
+      }};
+    var renderer = jsTestUtils.shallowRender(
+      <juju.components.AddedServicesListItem
+        focusService={sinon.stub()}
+        unfocusService={sinon.stub()}
+        fadeService={sinon.stub()}
+        unfadeService={sinon.stub()}
+        changeState={sinon.stub()}
+        getUnitStatusCounts={sinon.stub()}
+        service={service} />, true);
+
+    var instance = renderer.getMountedInstance();
+    assert.equal(instance.state.focus, true);
+    assert.equal(instance.state.fade, true);
+
+    // This is ugly but we have to check that the proper name prop was passed
+    // to the SvgIcon component.
+    var output = renderer.getRenderOutput();
+    assert.deepEqual(
```

Comment:
Yeah, this test in particular was quite a struggle because I have to use shallowRender so that the sub components don't render but I also wanted to make sure that the proper props are passed to the sub component so the only way I could figure that out is by doing that deep object check